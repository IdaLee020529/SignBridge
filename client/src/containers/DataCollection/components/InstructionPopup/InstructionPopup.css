.instruction-popup {
    width: 1000px;
    height: 600px;
    background-color: #f9f9f9;
    border: 1px solid #ccc;
    border-radius: 15px;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
    font-size: 18px;
    color: #333;
    padding: 30px 40px 40px 40px;
    z-index: 1001;
    position: absolute;
    display: none;
}


.instruction-popup.shown {
    display: block;
}

.instruction-popup-header {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.instruction-title {
    margin: 0 auto;
}

.instruction-popup-details {
    height: 90%;
    overflow-y: auto;
    display: flex;
    flex-direction: column;
    margin-top: 10px;

    &::-webkit-scrollbar {
        width: 10px;
    }

    &::-webkit-scrollbar-thumb {
        background: #ccc;
    }

    &::-webkit-scrollbar-thumb:hover {
        background: #888;
    }
}

.line-separator {
    width: 95%;
    height: 1px;
    background-color: #ccc;
    margin: 20px 0;
}

.instruction-popup-details {
    margin-top: 30px;
}

.instruction-popup-details-section1 p {
    font-size: 20px;
}

.instruction-popup-details-section1 p strong {
    font-size: 26px;
}


.instruction-popup-details-section2 p {
    font-size: 26px;
    font-weight: bold;

}

.instruction-popup-content {
    padding: 10px 20px;
    /* Add padding to the content area */
}

.instruction-popup-content ul {
    list-style-type: disc;
    /* Use disc bullets */
}

.instruction-popup-content li {
    margin-bottom: 5px;
    /* Add space between list items */
}

.instruction-popup-details-image {
    display: flex;
    justify-content: center;
    /* Horizontally center the image */
    align-items: center;
    /* Vertically center the image */
    margin: 30px auto;
}

.instruction-image {
    width: 90%;
    height: auto;
}

.close-icon {
    cursor: pointer;
}

.close-icon:hover {
    color: red;
    /* Change color to red on hover */
}