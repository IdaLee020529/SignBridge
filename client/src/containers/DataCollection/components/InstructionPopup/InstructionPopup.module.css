.noScroll {
    overflow: hidden;
}

.instruction-popup {
    width: 1000px;
    height: 600px;
    background-color: #f9f9f9;
    border: 1px solid #ccc;
    border-radius: 15px;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
    font-size: 18px;
    color: #333;
    padding: 30px 40px 40px 40px;
    z-index: 1001;
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    display: none;
}

.instruction-popup.shown {
    display: block;
}

.instruction_popup_header {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.instruction_title {
    margin: 0 auto;
}

.instruction_popup_details {
    height: 90%;
    overflow-y: auto;
    display: flex;
    flex-direction: column;
    margin-top: 10px;

    &::-webkit-scrollbar {
        width: 10px;
    }

    &::-webkit-scrollbar-thumb {
        background: #ccc;
    }

    &::-webkit-scrollbar-thumb:hover {
        background: #888;
    }
}

.line_separator {
    width: 95%;
    height: 1px;
    background-color: #ccc;
    margin: 20px 0;
}

.instruction_popup_details {
    margin-top: 30px;
}

.instruction_popup_details_section1 p {
    font-size: 20px;
}

.instruction_popup_details_section1 p strong {
    font-size: 26px;
}


.instruction_popup_details_section2 p {
    font-size: 26px;
    font-weight: bold;

}

.instruction_popup_content {
    padding: 10px 20px;
    /* Add padding to the content area */
}

.instruction_popup_content ul {
    list-style-type: disc;
    /* Use disc bullets */
}

.instruction_popup_content li {
    margin-bottom: 5px;
    /* Add space between list items */
}

.instruction_popup_details_image {
    display: flex;
    justify-content: center;
    /* Horizontally center the image */
    align-items: center;
    /* Vertically center the image */
    margin: 30px auto;
}

.instruction_image {
    width: 90%;
    height: auto;
}

.fa {
    float: right;
    font-size: 36px;
    cursor: pointer;
}

.fa:hover {
    text-shadow: 0 4px 8px rgba(0, 0, 0, 0.5);
}