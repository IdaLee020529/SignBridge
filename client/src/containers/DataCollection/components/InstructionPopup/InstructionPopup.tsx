import React from "react";
import { CloseOutlined } from "@ant-design/icons";
import "./InstructionPopup.css";
import InstructionImage from "/images/InfoPopup.png";

interface InstructionPopupProps {
  showInstructionPopup: boolean;
  onClose: () => void;
}

const InstructionPopup: React.FC<InstructionPopupProps> = ({
  showInstructionPopup,
  onClose,
}) => {
  return (
    <div
      className={`instruction-popup ${showInstructionPopup ? "shown" : ""} `}
    >
      <div className="instruction-popup-header">
        <h1 className="instruction-title">Instructions</h1>
        <i className={`fa fa fa-close`} onClick={onClose}></i>
      </div>
      <div className="instruction-popup-details">
        <div className="instruction-popup-details-section1">
          <p>
            <strong>1.</strong> Login a user account
          </p>
          <p>
            <strong>2.</strong> Fill up the details - Name, Email, Text/
            Sentence
          </p>
          <p>
            <strong>3.</strong> Upload a demonstration video of the
            text/sentence
          </p>
          <p>
            <strong>4.</strong> Press the reset button to reset the form or the
            submit button to submit the form
          </p>
        </div>
        <hr className="line-separator" />
        <div className="instruction-popup-details-section2">
          <p>Note:</p>
          <div className="instruction-popup-content">
            <ul>
              <li>The video submitted must be in mp4 format</li>
              <li>
                The video should include either half or entire body of the
                presenter
              </li>
              <li>
                The video will only be used for avatar preparation purpose only
              </li>
            </ul>
          </div>
        </div>
        <div className="instruction-popup-details-section3">
          <p>You can refer to the image below:</p>
          <div className="instruction-popup-details-image">
            <img
              src={InstructionImage}
              alt="instruction-image"
              className="instruction-image"
            />
          </div>
        </div>
      </div>
    </div>
  );
};

export default InstructionPopup;
