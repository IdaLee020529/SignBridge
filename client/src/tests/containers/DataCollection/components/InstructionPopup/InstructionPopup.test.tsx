import React from "react";
import InstructionPopup from "../../../../../containers/DataCollection/components/InstructionPopup/InstructionPopup";
import { cleanup } from "@testing-library/react";
import { create } from "react-test-renderer";

describe("Test InstructionPopup", () => {
  afterEach(cleanup); //Unmount all react tree after rendering
  const props = {
    showInstructionPopup: true,
    onClose: jest.fn(),
  };
  it("should render correctly", () => {
    const tree = create(<InstructionPopup {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
