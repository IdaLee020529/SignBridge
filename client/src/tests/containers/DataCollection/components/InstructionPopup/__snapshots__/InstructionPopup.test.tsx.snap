// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InstructionPopup should render correctly 1`] = `
<div
  className="instruction-popup shown "
>
  <div
    className="instruction-popup-header"
  >
    <h1
      className="instruction-title"
    >
      Instructions
    </h1>
    <span
      aria-label="close"
      className="anticon anticon-close close-icon"
      onClick={[MockFunction]}
      role="img"
      tabIndex={-1}
    >
      <svg
        aria-hidden="true"
        data-icon="close"
        fill="currentColor"
        fillRule="evenodd"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M799.86 166.31c.02 0 .04.02.08.06l57.69 57.7c.04.03.05.05.06.08a.12.12 0 010 .06c0 .03-.02.05-.06.09L569.93 512l287.7 287.7c.04.04.05.06.06.09a.12.12 0 010 .07c0 .02-.02.04-.06.08l-57.7 57.69c-.03.04-.05.05-.07.06a.12.12 0 01-.07 0c-.03 0-.05-.02-.09-.06L512 569.93l-287.7 287.7c-.04.04-.06.05-.09.06a.12.12 0 01-.07 0c-.02 0-.04-.02-.08-.06l-57.69-57.7c-.04-.03-.05-.05-.06-.07a.12.12 0 010-.07c0-.03.02-.05.06-.09L454.07 512l-287.7-287.7c-.04-.04-.05-.06-.06-.09a.12.12 0 010-.07c0-.02.02-.04.06-.08l57.7-57.69c.03-.04.05-.05.07-.06a.12.12 0 01.07 0c.03 0 .05.02.09.06L512 454.07l287.7-287.7c.04-.04.06-.05.09-.06a.12.12 0 01.07 0z"
        />
      </svg>
    </span>
  </div>
  <div
    className="instruction-popup-details"
  >
    <div
      className="instruction-popup-details-section1"
    >
      <p>
        <strong>
          1.
        </strong>
         Login a user account
      </p>
      <p>
        <strong>
          2.
        </strong>
         Fill up the details - Name, Email, Text/ Sentence
      </p>
      <p>
        <strong>
          3.
        </strong>
         Upload a demonstration video of the text/sentence
      </p>
      <p>
        <strong>
          4.
        </strong>
         Press the reset button to reset the form or the submit button to submit the form
      </p>
    </div>
    <hr
      className="line-separator"
    />
    <div
      className="instruction-popup-details-section2"
    >
      <p>
        Note:
      </p>
      <div
        className="instruction-popup-content"
      >
        <ul>
          <li>
            The video submitted must be in mp4 format
          </li>
          <li>
            The video should include either half or entire body of the presenter
          </li>
          <li>
            The video will only be used for avatar preparation purpose only
          </li>
        </ul>
      </div>
    </div>
    <div
      className="instruction-popup-details-section3"
    >
      <p>
        You can refer to the image below:
      </p>
      <div
        className="instruction-popup-details-image"
      >
        <img
          alt="instruction-image"
          className="instruction-image"
          src="default"
        />
      </div>
    </div>
  </div>
</div>
`;
